Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Call.hs

```diff
@@ -50,13 +51,27 @@ directCallStatusCallback callSid dialCallStatus recordingUrl_ callDuration = do
   runTransaction $ QCallStatus.updateCallStatus callStatus.id (exotelStatusToInterfaceStatus dialCallStatus) (fromMaybe 0 callDuration) recordingUrl
   return Ack
 
-getCustomerMobileNumber :: (CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, EncFlow m r) => Text -> Text -> Text -> ExotelCallStatus -> m GetCustomerMobileNumberResp
-getCustomerMobileNumber callSid callFrom_ callTo_ callStatus = do
+makeEnteredDtmfFlowKey :: Text -> Text
+makeEnteredDtmfFlowKey callSid = "DriverEnteredDtmfFlow:CallSid-" <> callSid
+
+getCustomerMobileNumber :: (Redis.HedisFlow m r, CacheFlow m r, EsqDBFlow m r, EsqDBReplicaFlow m r, EncFlow m r) => Text -> Text -> Text -> Text -> ExotelCallStatus -> m GetCustomerMobileNumberResp
+getCustomerMobileNumber callSid callFrom_ callTo_ digits callStatus = do
   let callFrom = dropFirstZero callFrom_
   let callTo = dropFirstZero callTo_
   mobileNumberHash <- getDbHash callFrom
+  dtmfMobileHash <- getDbHash digits
   exophone <- CQExophone.findByPhone callTo >>= fromMaybeM (ExophoneDoesNotExist callTo)
-  driver <- runInReplica $ QPerson.findByMobileNumberAndMerchant "+91" mobileNumberHash exophone.merchantId >>= fromMaybeM (PersonWithPhoneNotFound callFrom)
+  let enteredDtmfFlow = False
+  expTime <- fromIntegral <$> asks (.cacheConfig.configsExpTime)
+  let enteredDtmfFlowKey = makeEnteredDtmfFlowKey callSid
+  Redis.setExp enteredDtmfFlowKey enteredDtmfFlow expTime
```

Comment:
Why to set redis key in this flow?

Reply:
This I have used as a flag if it is false means driver has not entered dtmf flow

Reply:
are you using this stored info in some other API ? if its used only in the scope of this function then there shouldn't be a need to set in redis.

Reply:
no using . removed redis flow